src/Microsoft.AspNetCore.OData/ODataOptions.cs

```
@@ -129,14 +129,22 @@ public ODataOptions AddRouteComponents(string routePrefix, IEdmModel model, Acti
                 throw Error.ArgumentNull(nameof(model));
             }
 
-            if (RouteComponents.ContainsKey(routePrefix))
+            if (routePrefix == null)
             {
-                throw Error.InvalidOperation(SRResources.ModelPrefixAlreadyUsed, routePrefix);
+                throw Error.ArgumentNull(nameof(routePrefix));
             }
 
+            string sanitizedRoutePrefix = routePrefix.Trim('/');
+
+            if (RouteComponents.ContainsKey(sanitizedRoutePrefix))
+            {
+                throw Error.InvalidOperation(SRResources.ModelPrefixAlreadyUsed, sanitizedRoutePrefix);
+            }
+
+
             // Consider to use Lazy<IServiceProvider> ?
             IServiceProvider serviceProvider = BuildRouteContainer(model, configureServices);
-            RouteComponents[routePrefix] = (model, serviceProvider);
+            RouteComponents[sanitizedRoutePrefix] = (model, serviceProvider);
```

I'm debating in my mind whether it's better to throw an exception if the user includes the slashes in the route prefix or cure the problem and move on... What was the behaviour in 7.x?

---

Based on a previous triage discussion. It was agreed to just sanitize the slashes. If we don't sanitize the slashes then it should be clearly documented that the prefix should not contain slashes.

---

The `MapODataRoute` method in 7.x removes the last trailing slash. I doesn't appear like it removes leading slashes.

---

I agree with @habbes. We should just sanitize the slashes instead of throwing an exception.